server/planarserver.py

```diff
@@ -129,16 +122,140 @@ async def start_servers():
     print()
 
 
-main_app.on_shutdown.append(on_shutdown)
+def server_main(_args):
+    """Start the PlanarAlly server."""
 
+    if (not (FILE_DIR / "templates").exists()) and ("dev" not in sys.argv):
+        print(
+            "You must gather your par— you must build the client, before starting the server.\nSee https://www.planarally.io/server/setup/self-hosting/ on how to build the client or import a pre-built client."
+        )
+        sys.exit(1)
 
-if __name__ == "__main__":
+    mimetypes.init()
+    mimetypes.types_map[".js"] = "application/javascript; charset=utf-8"
+
+    save.check_save()
+
+    loop = asyncio.get_event_loop()
     loop.create_task(start_servers())
 
     try:
+        main_app.on_shutdown.append(on_shutdown)
+
         loop.run_forever()
     except:
         pass
     finally:
         for runner in runners:
             loop.run_until_complete(runner.cleanup())
+
+
+resource_types = [User, Room]
+
+
+def list_main(args):
+    """List all of the requested resource type."""
+    for resource_type in resource_types:
+        if resource_type.__name__.lower() == args.resource:
+            for resource in resource_type.select():
+                print(resource.name)
+
+
+def remove_main(args):
+    """Remove a requested resource."""
+    for resource_type in resource_types:
+        if resource_type.__name__.lower() == args.resource:
+            chosen_resource = resource_type.get_or_none(name=args.name)
+            chosen_resource.delete_instance()
+
+
+def reset_password_main(args):
+    """Reset a users password. Will prompt for the new password if not provided."""
+    password = args.password
+    if not password:
+        first_password = getpass.getpass()
+        second_password = getpass.getpass("Retype password:")
+        while first_password != second_password:
+            print("Passwords do not match.")
+            first_password = getpass.getpass()
+            second_password = getpass.getpass("Retype password:")
+        password = first_password
+    user = User.by_name(args.name)
```

Comment:
Maybe fetch the user first before doing the password prompts.
If the user does not exist (e.g. typo) then typing the password twice was not in vain.